components/layout/layout_task.rs

```diff
@@ -1395,6 +1420,38 @@ impl FragmentBorderBoxIterator for CollectingFragmentBorderBoxIterator {
     }
 }
 
+struct FragmentLocatingFragmentIterator {
+    node_address: OpaqueNode,
+    client_rect: Rect<i32>,
+}
+
+impl FragmentLocatingFragmentIterator {
+    fn new(node_address: OpaqueNode) -> FragmentLocatingFragmentIterator {
+        FragmentLocatingFragmentIterator {
+            node_address: node_address,
+            client_rect: Rect::zero()
+        }
+    }
+}
+
+impl FragmentBorderBoxIterator for FragmentLocatingFragmentIterator {
+    fn process(&mut self, fragment: &Fragment, border_box: &Rect<Au>) {
+        let border_style_struct = fragment.style.get_border();
+        let top_width = border_style_struct.border_top_width;
+        let right_width = border_style_struct.border_right_width;
+        let bottom_width = border_style_struct.border_bottom_width;
+        let left_width = border_style_struct.border_left_width;
```

Comment:
(Just for future reference, this could be a single let statement: `let style_structs::Border { border_top_width: top_width, border_right_width: right_width, ... } = *fragment.style.get_border();` It's fine the way it is though.)